Index: path.cpp
===================================================================
--- path.cpp	(revision 861897)
+++ path.cpp	(working copy)
@@ -223,6 +223,16 @@
     }
 }
 
+MediaNode *Path::sourceNode() const
+{
+    return d->sourceNode;
+}
+
+MediaNode *Path::sinkNode() const
+{
+    return d->sinkNode;
+}
+
 bool Path::disconnect()
 {
     if (!isValid()) {
Index: path.h
===================================================================
--- path.h	(revision 861897)
+++ path.h	(working copy)
@@ -184,6 +184,16 @@
         bool reconnect(MediaNode *source, MediaNode *sink);
 
         /**
+         * Returns the source MediaNode used by the path.
+         */
+        MediaNode *sourceNode() const;
+
+        /**
+         * Returns the sink MediaNode used by the path.
+         */
+        MediaNode *sinkNode() const;
+
+        /**
          * Disconnects the path from the MediaNodes it was connected to. This invalidates the path
          * (isValid returns \p false then).
          */
