Index: FindQt4.cmake
===================================================================
--- FindQt4.cmake	(revision 302)
+++ FindQt4.cmake	(working copy)
@@ -1273,7 +1273,7 @@
       FOREACH(_ts_file ${_my_tsfiles})
         ADD_CUSTOM_COMMAND(OUTPUT ${_ts_file}
            COMMAND ${QT_LUPDATE_EXECUTABLE}
-           ARGS ${_my_sources} -ts ${_ts_file}
+           ARGS -noobsolete ${_my_sources} -ts ${_ts_file}
            DEPENDS ${_my_sources})
       ENDFOREACH(_ts_file)
       QT4_ADD_TRANSLATION(${_qm_files} ${_my_tsfiles})
@@ -1287,7 +1287,7 @@
 
          ADD_CUSTOM_COMMAND(OUTPUT ${qm}
             COMMAND ${QT_LRELEASE_EXECUTABLE}
-            ARGS ${_abs_FILE} -qm ${qm}
+            ARGS -compress -nounfinished -removeidentical ${_abs_FILE} -qm ${qm}
             DEPENDS ${_abs_FILE}
          )
          SET(${_qm_files} ${${_qm_files}} ${qm})
